Sofa/Component/SolidMechanics/Spring/src/sofa/component/solidmechanics/spring/RestShapeSpringsForceField.inl

```diff
@@ -496,4 +497,37 @@ void RestShapeSpringsForceField<DataTypes>::addKToMatrix(const MechanicalParams*
     }
 }
 
+template<class DataTypes>
+void RestShapeSpringsForceField<DataTypes>::buildStiffnessMatrix(core::behavior::StiffnessMatrix* matrices)
+{
+    const VecReal& k = d_stiffness.getValue();
+    constexpr sofa::Size N = Coord::total_size;
+
+    auto dfdx = matrices->getForceDerivativeIn(this->mstate.get())
+                         .withRespectToPositionsIn(this->mstate.get());
+
+    if (k.size() != m_indices.size())
+    {
+        const Real k0 = -k[0];
+
+        for (const auto index : m_indices)
+        {
+            for(sofa::Index i = 0; i < N; i++)
+            {
+                dfdx(N * index + i, N * index + i) += k0;
+            }
+        }
+    }
+    else
+    {
+        for (const auto index : m_indices)
+        {
+            for(sofa::Index i = 0; i < N; i++)
+            {
+                dfdx(N * index + i, N * index + i) += - k[index];
+            }
+        }
+    }
```

Comment:
One problematic case:
 - k has only one stiffness (one for all for example)
 - m_indices has a size of one (only one index)
 
k[index] will go out of bound if index != 0

Reply:
Ah yes, I'll fix that. Thanks